skimage/measure/_marching_cubes_lewiner.py

```diff
@@ -0,0 +1,204 @@
+import sys
+import base64
+
+import numpy as np
+
+if sys.version_info[0] == 3:
+    base64decode = base64.decodebytes
+else:
+    base64decode = base64.decodestring
+
+from . import _marching_cubes_lewiner_luts as mcluts
+from . import _marching_cubes_lewiner_cy
+
+
+
+def marching_cubes_lewiner(volume, level=None, spacing=(1., 1., 1.),
+                           gradient_direction='descent', step_size=1,
+                           allow_degenerate=True, use_classic=False):
+    """
+    Lewiner marching cubes algorithm to find surfaces in 3d volumetric data
+    
+    In contrast to ``marching_cubes()``, this algorithm resolves
+    ambiguities and guarantees topologically correct results.
+    
+    Parameters
+    ----------
+    volume : (M, N, P) array
+        Input data volume to find isosurfaces. Will internally be
+        converted to float32 if necessary.
+    level : float
+        Contour value to search for isosurfaces in `volume`. If not
+        given or None, the average of the min and max of vol is used.
+    spacing : length-3 tuple of floats
+        Voxel spacing in spatial dimensions corresponding to numpy array
+        indexing dimensions (M, N, P) as in `volume`.
+    gradient_direction : string
+        Controls if the mesh was generated from an isosurface with gradient
+        descent toward objects of interest (the default), or the opposite,
+        considering the *left-hand* rule.
+        The two options are:
+        * descent : Object was greater than exterior
+        * ascent : Exterior was greater than object
+    step_size : int
+        Step size in voxels. Default 1. Larger steps yield faster but
+        coarser results. The result will always be topologically correct
+        though.
+    allow_degenerate : bool
```

Comment:
It takes between 10% and 50% of extra time. That's a lot of overhead, especially since the output might be the same, since degenerate triangles are rare in real world data,